Support App, Lsig by names (#570)
scale-it/algo-builder: docs/guide/deployer.md

You can learn more about Stateless Smart Contracts [here](https://developer.algorand.org/docs/features/asc1/stateless/).

#### Checkpoint names

> Storing data in [checkpoint](https://algobuilder.dev/guide/execution-checkpoints.html) against your own names can be very useful. For example, you don't need to pass smart contract template parameters every time while getting app info, or loading a logic signature. Currently, we support naming for apps (Algorand stateful application) and logic signatures.

Reviewer: ```suggestion
Algob creates [checkpoint](https://algobuilder.dev/guide/execution-checkpoints.html) and associates them with a name. This is a very useful feature. For example, you don't need to pass smart contract template parameters every time when getting app info, or loading a logic signature. Since `algob v4.0`, we support naming for apps (Algorand stateful smart contracts) and smart signatures.
```

##### App Name

> You can pass `appName` in `deployer.deployApp` while deploying your application. The app metadata in checkpoint will be stored against "appName". Eg.

Reviewer: ```suggestion
`deployer.deployApp` requires `appName` when deploying an application. The app metadata in checkpoint will be stored against "appName". Eg.
```

  const bytecode = info.compiled;
  ```

  * `getDeployedASC`: Similar to above, but instead of compiling, it returns cached program (from artifacts/cache) by app/lsig name.
  ```js
  // lsig
  const info = await deployer.getDeployedASC('MyApp');
  const [approvalInfo, clearInfo] = [info.approval, info.clear];

  const lsigInfo = await deployer.getDeployedASC('MyLsig');

>   ```

Author: ok

#### Checkpoint names

> Storing data in checkpoint against your own names can be very useful. For example, you don't need to pass smart contract template parameters every time while getting app info, or loading a logic signature. Currently, we support naming for apps (Algorand stateful application) and logic signatures.

Author: ok
